gapis/api/vulkan/overdraw.go

```diff
@@ -0,0 +1,1322 @@
+	if !pInfo.MultisampleState().IsNil() {
+		info := pInfo.MultisampleState()
+		sampleMaskPtr, _ := unpackMapMaybeEmpty(info.SampleMask())
+		multisamplePtr = allocAndRead(
+			NewVkPipelineMultisampleStateCreateInfo(a,
+				VkStructureType_VK_STRUCTURE_TYPE_PIPELINE_MULTISAMPLE_STATE_CREATE_INFO, // sType
+				0, // pNext
+				0, // flags
+				info.RasterizationSamples(),
+				info.SampleShadingEnable(),
+				info.MinSampleShading(),
+				NewVkSampleMaskᶜᵖ(sampleMaskPtr), // pSampleMask
+				info.AlphaToCoverageEnable(),
+				info.AlphaToOneEnable(),
+			)).Ptr()
+	}
+
+	var depthStencilPtr memory.Pointer
+	{
+		// FIXME: work with existing depth buffer
+		stencilOp := NewVkStencilOpState(a,
+			0, // failOp
+			VkStencilOp_VK_STENCIL_OP_INCREMENT_AND_CLAMP, // passOp
+			0, // depthFailOp
+			VkCompareOp_VK_COMPARE_OP_ALWAYS, // compareOp
+			255, // compareMask
+			255, // writeMask
+			0,   // reference
+		)
+		state := MakeVkPipelineDepthStencilStateCreateInfo(a)
+		state.SetSType(
+			VkStructureType_VK_STRUCTURE_TYPE_PIPELINE_DEPTH_STENCIL_STATE_CREATE_INFO)
+		state.SetStencilTestEnable(1)
+		state.SetFront(stencilOp)
+		state.SetBack(stencilOp)
+		if !pInfo.DepthState().IsNil() {
+			info := pInfo.DepthState()
+			if info.StencilTestEnable() != 0 {
+				return NilVkGraphicsPipelineCreateInfo,
+					fmt.Errorf("The stencil buffer is already in use")
+			}
+
+			state.SetDepthTestEnable(info.DepthTestEnable())
+			state.SetDepthWriteEnable(info.DepthWriteEnable())
+			state.SetDepthCompareOp(info.DepthCompareOp())
+			state.SetDepthBoundsTestEnable(info.DepthBoundsTestEnable())
+			state.SetMinDepthBounds(info.MinDepthBounds())
+			state.SetMaxDepthBounds(info.MaxDepthBounds())
+		}
+		depthStencilPtr = allocAndRead(state).Ptr()
+	}
+
+	colorBlendPtr := memory.Nullptr
+	if !pInfo.ColorBlendState().IsNil() {
+		info := pInfo.ColorBlendState()
+		attachmentPtr, attachmentCount := unpackMapMaybeEmpty(info.Attachments())
+		colorBlendPtr = allocAndRead(
+			NewVkPipelineColorBlendStateCreateInfo(a,
+				VkStructureType_VK_STRUCTURE_TYPE_PIPELINE_COLOR_BLEND_STATE_CREATE_INFO, // sType
+				0, // pNext
+				0, // flags
+				info.LogicOpEnable(),
+				info.LogicOp(),
+				attachmentCount, // attachmentCount
+				NewVkPipelineColorBlendAttachmentStateᶜᵖ(
+					attachmentPtr), // pAttachments
+				info.BlendConstants(),
+			)).Ptr()
+	}
+
+	dynamicPtr := memory.Nullptr
+	if !pInfo.DynamicState().IsNil() {
+		info := pInfo.DynamicState()
+		statesPtr, statesCount := unpackMapMaybeEmpty(info.DynamicStates())
+		dynamicPtr = allocAndRead(
+			NewVkPipelineDynamicStateCreateInfo(a,
+				VkStructureType_VK_STRUCTURE_TYPE_PIPELINE_DYNAMIC_STATE_CREATE_INFO, // sType
+				0,                              // pNext
+				0,                              // flags
+				statesCount,                    // dynamicStateCount
+				NewVkDynamicStateᶜᵖ(statesPtr), // pDynamicStates
+			)).Ptr()
+	}
+
+	// TODO: determine if basePipelineHandle is an issue
```

Comment:
You should be able to just pass in -1 / VK_NULL_HANDLE for base_pipeline_handle / base_pipeline_index.

Actually if you are passing in 0 for flags you MUST have -1 / VK_NULL_HANDLE for base_pipeline_*.

Alternatively, for a slight optimization, IF the original pipeline was created with `VK_PIPELINE_CREATE_ALLOW_DERIVATIVES_BIT` you could actaully set `VK_PIPELINE_CREATE_DERIVATIVE_BIT`, and point base_pipeline_handle to the original pipeline. That would improve pipeline compilation speed.